packages/advanced-logic/src/extensions/payment-network/reference-based.ts

```diff
@@ -91,4 +123,47 @@ export default abstract class ReferenceBasedPaymentNetwork<
       version: this.currentVersion,
     };
   }
+
+  /**
+   * Applies declare payment
+   *
+   * @param extensionState previous state of the extension
+   * @param extensionAction action to apply
+   * @param requestState request state read-only
+   * @param actionSigner identity of the signer
+   *
+   * @returns state of the updated extension
+   */
+  protected applyDeclarePayment(
+    extensionState: ExtensionTypes.IState,
+    extensionAction: ExtensionTypes.IAction,
+    requestState: RequestLogicTypes.IRequest,
+    actionSigner: IdentityTypes.IIdentity,
+    timestamp: number,
+  ): ExtensionTypes.IState {
+    if (!requestState.payee) {
+      throw Error(`The request must have a payee`);
+    }
+    if (!Utils.identity.areEqual(actionSigner, requestState.payee)) {
+      throw Error(`The signer must be the payee`);
+    }
+
+    const copiedExtensionState: ExtensionTypes.IState = Utils.deepCopy(extensionState);
+
+    // update the amount and txHash
+    copiedExtensionState.values.amount = extensionAction.parameters.amount;
```

Comment:
You should also validate the amount before using it. (cf. `applyDeclareReceivedPayment` in `declarative.ts`). Don't forget it's an open network so mistakes can be pushed by other nodes.

Once the amount is validated, the extensionState amount update should look more like `declarative.ts`:
```
    // increment receivedPaymentAmount
    copiedExtensionState.values.receivedPaymentAmount = Utils.amount.add(
      copiedExtensionState.values.receivedPaymentAmount,
      extensionAction.parameters.amount,
    );
```